### PR TITLE
bpo-46659: Fix the MBCS codec alias on Windows

### PR DESCRIPTION
**Pull request by vstinner**
https://bugs.python.org/issue46659

### CODE REVIEW

**Comment by vstinner** [comment]
@eryksun: Does this approach look good to you? It's basically restores Python 3.10 behavior. I just adjusted the test.

File: Lib/test/test_codecs.py [hunk 1]
```diff
@@ -3191,13 +3191,16 @@ def test_incremental(self):
...
-        # On Windows, the encoding name must be the ANSI code page
-        encoding = locale.getpreferredencoding(False)
-        self.assertTrue(encoding.startswith('cp'), encoding)
-
-        # The encodings module create a "mbcs" alias to the ANSI code page
-        codec = codecs.lookup(encoding)
-        self.assertEqual(codec.name, "mbcs")
+        # Check that looking up our 'default' codepage will return
+        # mbcs when we don't have a more specific one available
+        code_page = 9999
```
Comment (eryksun) [inline Lib/test/test_codecs.py]: Let's use 99999 instead, which is beyond the range of anything that Python might ever define. See [code page identifiers](https://docs.microsoft.com/en-us/windows/win32/intl/code-page-identifiers).

**Comment by eryksun** [comment]
> Does this approach look good to you? It's basically restores Python 3.10 behavior. I just adjusted the test.

Yes, having a fallback for the process ANSI code page is for the best -- even though we support all of the known code pages in standard Windows locales.  Thank you for clarifying the reason for its existence in the comments.

I'd prefer to also have a fallback for the process OEM code page, but adding `_winapi.GetOEMCP()` is an enhancement request.

**Comment by vstinner** [comment]
Thanks @eryksun for your review.

My use case is to deprecate the locale.getdefaultlocale() function. Adding an alias for the OEM page code is unrelated. If you consider that it's an useful feature, you can open a separated PR. So far, nobody requested it, so maybe it's not needed. I'm not convinced that the fallbak for the ANSI code page is used by anyone.

**Comment by eryksun** [comment]
> I'm not convinced that the fallbak for the ANSI code page is used by anyone.

I don't know whether the ANSI fallback is used by anyone. As I said, I've never worked with supplemental/replacement [custom locales](https://docs.microsoft.com/en-us/windows/win32/intl/custom-locales). The suggestion can be shelved until someone has an issue or explicitly requests a fallback for OEM as something required for their work environment.

**Comment by vstinner** [comment]
> The suggestion can be shelved until someone has an issue or explicitly requests a fallback for OEM as something required for their work environment.

It sounds perfectly reasonable :-)

**Pull request merged**